CUDADataFormats/SiPixelDigi/interface/SiPixelDigisCUDASOAView.h

```diff
@@ -0,0 +1,70 @@
+#ifndef CUDADataFormats_SiPixelDigi_interface_SiPixelDigisCUDASOAView_h
+#define CUDADataFormats_SiPixelDigi_interface_SiPixelDigisCUDASOAView_h
+
+#include <cuda_runtime.h>
+
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/host_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h"
+
+#include <cstdint>
+
+class SiPixelDigisCUDASOAView {
+public:
+  friend class SiPixelDigisCUDA;
+  friend class SiPixelRecHitSoAFromLegacy;
+  enum class StorageLocation {
+    kCLUS = 0,
+    kPDIGI = 2,
+    kRAWIDARR = 4,
+    kADC = 6,
+    kXX = 7,
+    kYY = 8,
+    kMODULEIND = 9,
+    kMAX = 10
+  };
+  /*
+  ============================================================================================================================
+  |          CLUS         |          PDIGI         |         RAWIDARR        |    ADC    |    XX     |     YY    | MODULEIND |
+  ============================================================================================================================
+  |        0: N*32        |         2: N*32        |         4: N*32         |  6: N*16  |  7: N*16  |  8: N*16  |  9: N*16  |
+  ============================================================================================================================
+  */
+  // These are for CPU output
+  // we don't copy local x and y coordinates and module index
+  enum class StorageLocationHost { kCLUS = 0, kPDIGI = 2, kRAWIDARR = 4, kADC = 6, kMAX = 7 };
+  /*
+  ========================================================================================
+  |          CLUS         |          PDIGI         |         RAWIDARR        |    ADC    |
+  ========================================================================================
+  |        0: N*32        |         2: N*32        |         4: N*32         |  6: N*16  |
+  ========================================================================================
+  */
+
+  __device__ __forceinline__ uint16_t xx(int i) const { return __ldg(xx_ + i); }
+  __device__ __forceinline__ uint16_t yy(int i) const { return __ldg(yy_ + i); }
+  __device__ __forceinline__ uint16_t adc(int i) const { return __ldg(adc_ + i); }
+  __device__ __forceinline__ uint16_t moduleInd(int i) const { return __ldg(moduleInd_ + i); }
+  __device__ __forceinline__ int32_t clus(int i) const { return __ldg(clus_ + i); }
+  __device__ __forceinline__ uint32_t pdigi(int i) const { return __ldg(pdigi_ + i); }
+  __device__ __forceinline__ uint32_t rawIdArr(int i) const { return __ldg(rawIdArr_ + i); }
+
+  uint16_t* xx() const { return xx_; }
```

Comment:
This is actually worse than the earlier version. Earlier gaining mutable access required copying the (`const`) View object, these give the access directly from `const` View. At minimum remove the `const`
```suggestion
  uint16_t* xx() { return xx_; }
```

Reply:
I added const accessors as well, for example to be used for copying, which does not require modification.
So there is 
```cpp
uint16_t* xx() { return xx_; }
// and
const uint16_t* xx() const { return xx_; }
```
now.